### PR TITLE
cpu/lpc11u34 : Peripheral addition : PWM

### PR DESCRIPTION
**Pull request by ks156**
This pull request add PWM capabilities to ```cpu/lpc11u34```.
It also add the configuration to use it in the ```board/weio```

### CODE REVIEW

File: boards/weio/include/periph_conf.h [hunk 1]
```diff
@@ -33,15 +33,15 @@ extern "C" {
...
 #define TIMER_0_MAX_VALUE   (0xffffffff)
-#define TIMER_0_CLKEN()     (LPC_SYSCON->SYSAHBCLKCTRL |= (1 << 9))
-#define TIMER_0_CLKDIS()    (LPC_SYSCON->SYSAHBCLKCTRL &= ~(1 << 9))
-#define TIMER_0_ISR         isr_ct32b0
-#define TIMER_0_IRQ         TIMER_32_0_IRQn
-/** @} */
+#define TIMER_0_CLKEN()     (LPC_SYSCON->SYSAHBCLKCTRL |= (1 << 10))
+#define TIMER_0_CLKDIS()    (LPC_SYSCON->SYSAHBCLKCTRL &= ~(1 << 10))
+#define TIMER_0_ISR         isr_ct32b1
+#define TIMER_0_IRQ         TIMER_32_1_IRQn
```
Comment (thomaseichinger) [inline boards/weio/include/periph_conf.h]: These timer changes seem unrelated to PWM. Do you mind opening a separate PR for it? Or keep the PR title more generic.

**Comment by thomaseichinger** [comment]
@ks156 Excellent work. I started Travis.

**Comment by thomaseichinger** [comment]
@ks156 Oh, and please squash.

File: boards/weio/include/periph_conf.h (hunk 1 above)
Comment (ks156) [inline boards/weio/include/periph_conf.h]: Actually, it is related to PWM. This MCU uses timers for PWM channels, and on the WeIO board, the PWM pins depend of CT16B0 and CT32B0.
Using CT32B0 as TIMER_0 and PWM in the same time will result of a concurrent use of the peripheral, and TIMER_0 will probably not work anymore once PWM_1 will be configured.

**Comment by ks156** [comment]
Squashed

**Comment by thomaseichinger** [comment]
@ks156 `test/periph_pwm` fails in the CI.

**Comment by ks156** [comment]
@thomaseichinger Indeed... I forgot to define PWM_0_CHANNELS and PWM_1_CHANNELS. Now it should be OK

**Comment by thomaseichinger** [comment]
Looks good, ACK when Travis passed.

**Comment by ks156** [comment]
I need to squash again once travis has passed ?

**Comment by thomaseichinger** [comment]
Good point! Sorry, overlooked this.
```C
if((new_str = malloc(strlen(my_comment_above) + strlen("and squashed")+1)) != NULL) {
    strcat(new_str, my_comment_above);
    strcat(new_str, " and squashed");
    puts(new_str);
}
```

**Comment by ks156** [comment]
@thomaseichinger Cortex M0 was OK now, so I squashed. Waiting for Travis now :smiley:

**Comment by thomaseichinger** [comment]
Squashed and Travis is happy. Go

**Comment by ks156** [comment]
Thanks @thomaseichinger. See you for the next PR :smile:
